connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -289,7 +285,6 @@ public void startConnector(
                 // Can't be put in a finally block because it needs to be swapped before the call on
                 // statusListener
                 Plugins.compareAndSwapLoaders(savedLoader);
-                workerMetricsGroup.recordConnectorStartupFailure();
```

Comment:
I think we might want to keep this line here in case we fail somehow before even creating the `WorkerConnector` instance.

Reply:
Good point. I believe we could achieve the same effect by wrapping the statusListener earlier in the method and using the wrapped listener in the exception handler. That way the only way we could fail without recording a metric would be if the wrapping itself failed (which is a dead simple constructor, so seems unlikely)

Reply:
Oooh, yeah that's neat. 👍